packages/jest-config/src/index.js

```diff
@@ -34,22 +39,22 @@ function readConfig(
   };
 }
 
-const parseConfig = argv =>
-  isJSONString(argv.config) ? JSON.parse(argv.config) : argv.config;
-
-const readOptions = (argv, root) => {
-  const rawOptions = parseConfig(argv);
-
-  if (typeof rawOptions === 'object') {
-    const config = Object.assign({}, rawOptions);
+const readOptions = (argv, root, dontUseArgvConfig) => {
+  // A JSON string was passed to `--config` argument and we can parse it
+  // and use as is.
+  if (isJSONString(argv.config)) {
+    const config = JSON.parse(argv.config);
     config.rootDir = config.rootDir || root;
     return config;
   }
 
-  if (typeof rawOptions === 'string') {
-    root = path.resolve(process.cwd(), rawOptions);
+  // A string passed to `--config`, which is either a direct path to the config
+  // or a path to directory containing `package.json` or `jest.conf.js`
+  if (!dontUseArgvConfig && typeof argv.config == 'string') {
```

Comment:
for MPR runs with `--config` specified this would always be true and `--config` value would've been taken for each project